### PR TITLE
Backport content access assignment for owner and distributors

### PR DESCRIPTION
**Pull request by wottop**

### CODE REVIEW

**Comment by wottop** [comment]
buildbot: retest this please

**Comment by Ceiu** [comment]
retest this please

File: server/src/main/java/org/candlepin/model/Owner.java [hunk 1]
```diff
@@ -397,4 +418,48 @@ public void setLogLevel(String logLevel) {
...
+     */
+    public String getContentAccessModeList() {
+        return contentAccessModeList;
+    }
+
+    public void setContentAccessModeList(String contentAccessModeList) {
+        this.contentAccessModeList = contentAccessModeList;
+    }
+
+    public String[] getContentAccessModeListAsArray() {
```
Comment (Ceiu) [inline server/src/main/java/org/candlepin/model/Owner.java]: Need a @JsonIgnore annotation here so this doesn't show up in the serialized output.

File: server/src/main/java/org/candlepin/resource/OwnerResource.java [hunk 2]
```diff
@@ -926,6 +926,28 @@ public Owner updateOwner(@PathParam("owner_key") @Verify(Owner.class) String key
             }
         }
 
+        // Update the contentAccess field if the incoming value is not null.
+        if (owner.getContentAccessMode() != null) {
+            StringTokenizer st = new StringTokenizer(toUpdate.getContentAccessModeList(), ",");
```
Comment (Ceiu) [inline server/src/main/java/org/candlepin/resource/OwnerResource.java]: This causes an NPE when updating an owner that doesn't yet have an access mode.

File: server/src/main/java/org/candlepin/resource/ConsumerResource.java [hunk 3]
```diff
@@ -903,6 +918,23 @@ protected boolean performConsumerUpdates(Consumer updated, Consumer toUpdate,
...
+            if (!StringUtils.isBlank(updated.getContentAccessMode()) &&
+                !ArrayUtils.contains(toUpdate.getOwner().getContentAccessModeListAsArray(),
+                    updated.getContentAccessMode())) {
+                throw new BadRequestException(i18n.tr(
+                    "The consumer cannot use the supplied content access mode."));
+            }
+            toUpdate.setContentAccessMode(updated.getContentAccessMode());
+            changesMade = true;
+        }
+        if (!StringUtils.isEmpty(updated.getContentAccessMode()) && !toUpdate.getType().isManifest()) {
```
Comment (Ceiu) [inline server/src/main/java/org/candlepin/resource/ConsumerResource.java]: This should be an alternate branch on the previous if-statement, I think.

File: server/src/main/java/org/candlepin/resource/OwnerResource.java [hunk 4]
```diff
@@ -926,6 +926,19 @@ public Owner updateOwner(@PathParam("owner_key") @Verify(Owner.class) String key
             }
         }
 
+        // Update the contentAccess field if the incoming value is not null.
+        if (owner.getContentAccessMode() != null) {
+            if (toUpdate.isAllowedContentAccessMode(owner.contentAccessMode())) {
+                String before = toUpdate.getContentAccessMode();
+                String after = owner.getContentAccessMode();
```
Comment (Ceiu) [inline server/src/main/java/org/candlepin/resource/OwnerResource.java]: before and after are never used here.

**Comment by Ceiu** [comment]
Looks good with the changes. Will ACK when the unused code is removed.

**Comment by Ceiu** [comment]
ACK

**Pull request merged**